src/Common/src/CoreLib/System/Collections/Generic/ValueListBuilder.cs

```diff
@@ -34,6 +36,13 @@ public void Append(T item)
             _pos = pos + 1;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public T Pop()
+        {
+            _pos--;
+            return _span[_pos];
+        }
```

Comment:
Do we need to add this to ValueListBuilder?  Seems like it could just be a helper in the regex code.

Reply:
I agree, I think this should be a dedicated type in Regex.